src/transformers/trainer_tf.py

```diff
@@ -118,18 +129,14 @@ def get_optimizers(
         If you want to use something else, you can pass a tuple in the Trainer's init,
         or override this method in a subclass.
         """
-        if self.optimizers is not None:
-            return self.optimizers
-
-        optimizer, scheduler = create_optimizer(
-            self.args.learning_rate,
-            self.train_steps,
-            self.args.warmup_steps,
-            adam_epsilon=self.args.adam_epsilon,
-            weight_decay_rate=self.args.weight_decay,
-        )
-
-        return optimizer, scheduler
+        if self.optimizers is None:
+            self.optimizers = create_optimizer(
+                self.args.learning_rate,
+                self.train_steps,
```

Comment:
Isn't it supposed to be the total number of train steps during the training, not only 1 epoch ? i.e. `(self.num_train_examples / self.args.train_batch_size) * self.args.`

Reply:
This is fixed in another PR, see #5051.

Reply:
So many PR x)